test_driver/app_test.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:flutter_driver/flutter_driver.dart';
+import 'package:test/test.dart';
+import 'package:flutter_survey/constant/widget_keys.dart';
+
+void main() {
+  group('Login Screen Test', () {
+    late FlutterDriver driver;
+
+    setUpAll(() async {
+      driver = await FlutterDriver.connect();
+      await driver.waitUntilFirstFrameRasterized();
+    });
+
+    tearDownAll(() async {
+      driver.close();
+    });
+
+    test(
+      'verify login error message',
+      () async {
+        String emailValue = 'abcd@gmail.com';
+        SerializableFinder emailTextField =
+            find.byValueKey(WidgetKeys.emailTextFieldKey);
+        await driver.waitFor(emailTextField);
+        await driver.tap(emailTextField);
+        await driver.enterText(emailValue);
+        SerializableFinder emailText = find.text(emailValue);
+        await driver.waitFor(emailText);
+
+        String passwordValue = '12345678910';
+        SerializableFinder passwordTextField =
+            find.byValueKey(WidgetKeys.passwordTextFieldKey);
+        await driver.waitFor(passwordTextField);
+        await driver.tap(passwordTextField);
+        await driver.enterText(passwordValue);
+        SerializableFinder passwordText = find.text(passwordValue);
+        await driver.waitFor(passwordText);
+
+        SerializableFinder loginButton =
+            find.byValueKey(WidgetKeys.loginButtonKey);
+        await driver.waitFor(loginButton);
+        await driver.tap(loginButton);
+
+        String errorMessage = 'Bad request';
+        SerializableFinder loginErrorText = find.text(errorMessage);
+        await driver.waitFor(loginErrorText);
+        expect(await driver.getText(loginErrorText), errorMessage);
+      },
+      timeout: const Timeout(
+        Duration(minutes: 2),
+      ),
+    );
```

Comment:
@Shayokh144 @Tuubz I'm curious why don't we use [Widget Test with integration_test_driver](https://docs.flutter.dev/cookbook/testing/widget/introduction) but [Flutter Driver via driver_extension](https://medium.com/flutter/integration-testing-with-flutter-driver-36f66ede5cf2)? I doubt using `Widget Test` would bring a shorter and simpler code https://docs.flutter.dev/cookbook/testing/integration/introduction#4-write-the-integration-test 💭

Reply:
Actually, from your PR's description 

> There are 2 options to run UI test Flutter's built-in testing framework and Flutter Driver. Flutter driver is mostly used so, I prefer this one.

I don't think Flutter Driver is more popular than Flutter's built-in testing framework. As there is migration documentation from flutter_driver here https://docs.flutter.dev/release/breaking-changes/flutter-driver-migration

More info here: https://medium.com/flutter/updates-on-flutter-testing-f54aa9f74c7e

> Unfortunately, Flutter driver tests have a number of issues. One issue is that tests run from the development machine and communicate with the app on the device, which means that the tests aren’t suitable for running on a device farm like [Firebase Test Lab](https://firebase.google.com/docs/test-lab). Another issue is that having a separate process for your tests makes it difficult to check the state of your app. You can check the output, but how do you know, for example, the internal state of the app. And, finally, the flutter_driver API is more complicated than it needs to be, particularly when it comes to writing the code to find the appropriate widgets on the screen to test against.

Reply:
@luongvo Hmmmmmm, good to know, however according to the requirements we must create UI tests, not integration tests. So I would say this is the correct approach, right? 💭 

Reference: https://nimblehq.co/compass/development/testing/#testing-pyramid

cc @Shayokh144

Reply:
@Tuubz Both Flutter Driver & Flutter Widget Test can be used for both UI & Integration testing. However, since Flutter Driver is deprecated, the migration to Flutter Widget Test is needed. Do you wanna work on deprecated technology?
- https://github.com/appium-userland/appium-flutter-driver/issues/210
- https://tomek-polanski.medium.com/the-new-flutter-integration-tests-are-they-any-good-a7c0fc506d6b

Reply:
Can't argue with that, let's refactor to "Flutter Widget Test" 🚀

Reply:
@Tuubz Please check out this documentation for more information https://docs.flutter.dev/testing
- https://docs.flutter.dev/cookbook/testing/widget/introduction
- https://docs.flutter.dev/testing/integration-tests

If you disagree with my recommendation, please feel free to raise to our guild channel to receive more ideas 🙏